incuna-sass/mixins/_grid.sass

```diff
@@ -0,0 +1,37 @@
+// Mixin to add a grid background to elements
+
+// Can be used to create a grid lines on their own or with a solid color background
+// Direction can be set to create just horizontal or just vertical lines, or left
+// as 'both' to create a grid
+
+// Example usage:
+// .element
+//     @include grid(4px, blue)
+// will create a transparent background with a grid that has 4px squares and 1px opaque
+// blue grid lines in both directions
+
+// .element
+//     @include grid(10px, red, 0.4, orange)
+// will create an orange background with a grid that has 10px squares and 1px red lines
+// at 40% opacity
+
+// .element
+//     @include grid(20px, green, 1, transparent, 5px, horizontal)
+// will create a transparent background with 5px opaque green lines going horizontally
+// across the page 20px apart
+
+@mixin grid($grid-size, $grid-color, $grid-opacity: 1, $bg-color: transparent, $line-thickness: 1px, $direction: both)
```

Comment:
This way you don't have to worry about order of arguments since required arguments must come before optional ones with defaults. You can then also have optional ones that default to `null` or "do nothing with this argument".

```sass
@mixin color-and-width($options: ())
    $defaults: (color: red, width: 10px)
    $options: map-merge($defaults, $options);

    $color: map-get($options, 'color')
    $width: map-get($options, 'width')
    $content: map-get($options, 'content') // null if not passed

    background-color: $color
    width: $width

    @if $content != null
        content: $content
```
```sass
.default
    @include color-and-width
.blue
    @include color-and-width((color: blue))
.content
    @include color-and-width((content: 'some content'))
```
```css
.default {
  background-color: red;
  width: 10px; }

.blue {
  background-color: blue;
  width: 10px; }

.content {
  background-color: red;
  width: 10px;
  content: "some content"; }
```

See http://sass-lang.com/documentation/file.SASS_REFERENCE.html#maps and http://sass-lang.com/documentation/Sass/Script/Functions.html#map-functions